src/autoscaler/sync/dblock.go

```diff
@@ -0,0 +1,79 @@
+package sync
+
+import (
+	"autoscaler/db"
+	"autoscaler/models"
+	"os"
+	"time"
+
+	"code.cloudfoundry.org/lager"
+
+	"github.com/tedsuo/ifrit"
+)
+
+type DatabaseLock struct {
+	logger lager.Logger
+}
+
+func NewDatabaseLock(logger lager.Logger) *DatabaseLock {
+	return &DatabaseLock{
+		logger: logger,
+	}
+}
+
+func (dblock *DatabaseLock) InitDBLockRunner(ttl time.Duration, owner string, lockDB db.LockDB) ifrit.Runner {
+	dbLockMaintainer := ifrit.RunFunc(func(signals <-chan os.Signal, ready chan<- struct{}) error {
+		lockTicker := time.NewTicker(ttl)
```

Comment:
should not we use LockRetryInterval instead of ttl ?

Reply:
we are renewing the lock every ttl timeframe. Database lock does not have any retry interval attribute.

Reply:
will this cause loss of the lock? I am assuming we need to refresh the lock less than ttl 
why not we add this? like consul lock?